### PR TITLE
Create a Core update site for Core plugins.

### PR DESCRIPTION
**Pull request by jadelkhoury**

### CODE REVIEW

File: org.eclipse.lyo.oslc4j.parent/pom.xml [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+	<pluginRepositories>
+		<pluginRepository>
+			<id>acceleo</id>
+			<url>https://repo.eclipse.org/content/repositories/acceleo-releases/</url>
+		</pluginRepository>
+	</pluginRepositories>
+	<modules>
+	<!--
+	-->
+		<module>../org.eclipse.lyo.repackaging.jersey</module>
```
Comment (yashkhatri) [inline org.eclipse.lyo.oslc4j.parent/pom.xml]: @jadelkhoury  If we compare it with [pom](https://github.com/eclipse/lyo.designer/blob/5c2b0d3f4fa009c248229010c9d35fc7f988e87f/pom.xml#L19) of lyo designer. Their modules are mention in the root pom and not the parent pom. Moreover, we need to specify a relative path here in the pom. I haven't tried cloning the repository and trying myself yet. However, look at this I guess you are getting errors that this  modules are not found?

File: org.eclipse.lyo.oslc4j.parent/pom.xml [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+						</environment>
+					</environments>
+				</configuration>
+			</plugin>
+		</plugins>
+		<pluginManagement>
+			<plugins>
+				<plugin>
+					<groupId>org.eclipse.acceleo</groupId>
+					<artifactId>org.eclipse.acceleo.maven</artifactId>
```
Comment (yashkhatri) [inline org.eclipse.lyo.oslc4j.parent/pom.xml]: Do we need acceleo-maven plugin in core?

**Comment by jadelkhoury** [comment]
I can actually add a pom.xml file to the feature project. This way, the feature project builds fine. But the update-site project complain that it cannot find the feature!
pom.xml file for the feature is:

**Comment by yashkhatri** [comment]
@jadelkhoury Yeah, I am working on it. Please add the pom file so that I can pull the latest changes.

**Comment by jadelkhoury** [comment]
It's in the latest comment above.
just create pom.xml file in the feature project. I did not want to add it (for now), since I want to show you the original error when one does not ahve it. (Why do we need it, when Lyo.designer features did not?). Something more fundamental seems wrong.

**Comment by berezovskyi** [comment]
I will take a look soon and will let you know

/Andrew



On Sat, Oct 13, 2018 at 3:32 PM +0200, "Jad El-khoury" <notifications@github.com<mailto:notifications@github.com>> wrote:


It's in the latest comment above.
just create pom.xml file in the feature project. I did not want to add it (for now), since I want to show you the original error when one does not ahve it. (Why do we need it, when Lyo.designer features did not?). Something more fundamental seems wrong.

**Comment by berezovskyi** [comment]
The new error I am having is

```
[ERROR]   Software being installed: core-osgi-updatesite raw:2.4.0.'SNAPSHOT'/format(n[.n=0;[.n=0;[-S]]]):2.4.0-SNAPSHOT
[ERROR]   Missing requirement: core-osgi-updatesite raw:2.4.0.'SNAPSHOT'/format(n[.n=0;[.n=0;[-S]]]):2.4.0-SNAPSHOT requires 'org.eclipse.lyo.core.osgi.core-osgi-jersey [2.4.0.SNAPSHOT]' but it could not be found
```

**Comment by jadelkhoury** [comment]
@berezovskyi so are we back to square one? Do we need yet another update-site for the Jersey bundle?
You seem to have done many changes, if you can give me a quick rundown, I can try to help out.

**Comment by jadelkhoury** [comment]
I believe you are getting an empty repo because your category.xml file of the update site does not seem to contain any Features, nor plugins. I got that error too until you have a feature and/or plugin.

Note also that we do not really need a Feature project. an update site can directly include a plugin.

Here are a couple of thinngs I noticed:

1. 
The pom.xml file under org.eclipse.lyo.core.osgi.bndl, at line 117 states:
`       <Bundle-Version>${project.version}</Bundle-Version>`
But that makes the plugin version "2.4.0-SNAPSHOT". We need "2.4.0.qualifier", right?
the manifest contains version "2.4.0.SNAPSHOT".  (Not it is even "." and not "-")

2. 
the pom.xml of the parent project has dependencies to org.eclipse.lyo.core.osgi & org.eclipse.lyo.core.osgi.
These are not necessary. I got less errors  when I removed these dependencies.

3.
Should the packageing for the bundle project be Jar? This created a whole bunch of new files.
`<packaging>jar</packaging>`

**Comment by berezovskyi** [comment]
What's the status on this?

**Comment by berezovskyi** [comment]
Fixed in LyoD

**Pull request closed without merging**